src/controllers/settings_controller.js

```diff
@@ -228,24 +230,103 @@ SettingsController.post('/skiptunnel', async (request, response) => {
   }
 });
 
+
 SettingsController.get('/tunnelinfo', async (request, response) => {
+  console.log('***** SettingsController.get: Tunnelinfo has been ' +
+              'called from the application *****');
   try {
+    // Check to see if we have a tunnel endpoint first
     const result = await Settings.get('tunneltoken');
+    let local_domain;
+    let mDNSstate;
+    let tunnelEndpoint;
+
     if (typeof result === 'object') {
-      const endpoint = `https://${result.name}.${
-        config.get('ssltunnel.domain')}`;
-      response.send(endpoint);
-      response.status(200).end();
+      console.log(`Tunnel domain found. Tunnel name is: ${result.name} and 
+      tunnel domain is: ${config.get('ssltunnel.domain')}`);
+      tunnelEndpoint = `https://${result.name}.
+      ${config.get('ssltunnel.domain')}`;
     } else {
-      response.status(404).end();
+      tunnelEndpoint = 'Not set.';
+    }
+
+    // Find out our default local DNS name Check for a previous name in the
+    // DB, if that does not exist use the default.
+    try {
+      mDNSstate = await Settings.get('multicastDNSstate');
+      local_domain = await Settings.get('localDNSname');
+    } catch (err) {
+      // Catch this DB error. Since we don't know what state the mDNS process
+      // should be in make sure it's off
+      console.error('Error getting DB entry for multicast from the DB: ${err}');
+      local_domain = config.get(
+        'settings.defaults.domain.local_control.mdns_service_domain');
     }
+
+    console.log(`Tunnel name is set to: ${tunnelEndpoint}`);
+    console.log(`Local mDNS Service Domain Name is: ${local_domain}`);
+    const local_domain_settings = {local_domain: local_domain,
+                                   mDNSstate: mDNSstate,
+                                   tunnelDomain: tunnelEndpoint};
+    response.send(local_domain_settings);
+    response.status(200).end();
   } catch (e) {
-    console.error('Failed to retrieve tunneltoken setting');
+    console.error('Failed to retrieve default settings for ' +
+      'tunneltoken or local service discovery setting');
     console.error(e);
     response.status(400).send(e);
   }
 });
 
+/* This is responsible for controlling dynamically the local domain name
+    settings (via mDNS) and changing or updating mozilla tunnel endpoints.
+    The /setdomain endpoint is invoked from:
+    MainMenu -> Settings -> Doamin
+
+    JSON data: { local: {
+                    multicastDNSstate: boolean,
+                    localDNSname: string, - e.g. MyHome
+                 mozillaTunnel: {
+                    tunnel: boolean,
+                    tunnelName: string, - e.g. MyName
+                    tunnelEmail: string
+                    }
+ */
+
+SettingsController.post('/setdomain', async (request, response) => {
```

Comment:
This would make more sense as `SettingsController.put('/domain', ...`

Reply:
done. more work than i thought! :-( ...